### PR TITLE
#1609 improve shunt regulation model allowing self and condensers tog…

### PR DESCRIPTION
**Pull request by marcochiaramello**
…ether

### CODE REVIEW

**Comment by marcochiaramello** [comment]
Closes #1609

**Review by gautierbureau — APPROVED** [review]

File: dynawo/sources/Models/CPP/Controls/Shunts/ModelCentralizedShuntsSectionControl/DYNModelCentralizedShuntsSectionControl.cpp [hunk 1]
```diff
@@ -178,131 +181,126 @@ namespace DYN {
...
-          break;
-        } else {
-          gLocal_[0] = ROOT_DOWN;
-          gLocal_[1] = ROOT_DOWN;
-          gLocal_[2] = ROOT_DOWN;
-          gLocal_[3] = ROOT_DOWN;
-        }
+    for (int s = 0; s < nbShunts_; ++s) {
+      if (isSelf_[s]) {
+          minValue = URefPu - deadBandsUPu_[s];
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/Controls/Shunts/ModelCentralizedShuntsSectionControl/DYNModelCentralizedShuntsSectionControl.cpp:186]: Could it be put outside the if ?

File: dynawo/sources/Models/CPP/Controls/Shunts/ModelCentralizedShuntsSectionControl/DYNModelCentralizedShuntsSectionControl.cpp [hunk 2]
```diff
@@ -178,131 +181,126 @@ namespace DYN {
...
+          maxValue = URefPu + deadBandsUPu_[s];
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/Controls/Shunts/ModelCentralizedShuntsSectionControl/DYNModelCentralizedShuntsSectionControl.cpp:187]: Same

File: nrt/data/DynaFlow/CentralizedShuntsRegulation/Shunt.crv [hunk 3]
```diff
@@ -18,12 +18,12 @@
   <curve model="Load__10" variable="load_UPu"/>
   <curve model="Generator___1" variable="generator_PGenPu"/>
```
Comment (cossartque) [inline nrt/data/DynaFlow/CentralizedShuntsRegulation/Shunt.crv:19]: It seems that the case is not fully at steady-state when the simulation starts.

**Review by cossartque — APPROVED** [review]

**Pull request merged**